feat: html reporter sharded option (#20737)
microsoft/playwright: docs/src/test-reporters-js.md

npx playwright show-report my-report
```

#### Sharded report

> When running tests on multiple [`--shards`](./test-parallel.md#shard-tests-between-multiple-machines), the `html` reporter can automatically show test results from all shards in one page. To achieve that pass `sharded: true` to the reporter configuration:

Reviewer: ```suggestion
When running tests on [multiple shards](./test-parallel.md#shard-tests-between-multiple-machines), the `html` reporter can automatically show test results from all shards in one page when configured with `sharded: true`.
```

> In configuration file, pass options directly:

Reviewer: ```suggestion
```

```js tab=js-js
// playwright.config.js
// @ts-check

const { defineConfig } = require('@playwright/test');

module.exports = defineConfig({
  reporter: [['html', { sharded: true }]],
});
```

```js tab=js-ts
// playwright.config.ts
import { defineConfig } from '@playwright/test';

export default defineConfig({
  reporter: [['html', { sharded: true }]],
});
```

> After the sharded tests finish, copy all files from `playwright-report` directory in every shard into a single directory. After that you can display the report by loading the `index.html` file from the folder or calling `npx playwright show-report` with the path to the folder.

Reviewer: ```suggestion
You can use sharded html report combined with a file hosting that allows serving html files.
In your CI recipe, after running tests in each shard, upload all files from `playwright-report` directory to the **same location**. After that you can open `index.html` from the uploaded location directly in the browser.
```
